packages/stark-ui/src/modules/minimap/components/minimap.component.ts

```diff
@@ -0,0 +1,173 @@
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import {
+	Component,
+	EventEmitter,
+	Inject,
+	Input,
+	NgZone,
+	OnChanges,
+	OnDestroy,
+	OnInit,
+	Output,
+	SimpleChanges,
+	ViewEncapsulation
+} from "@angular/core";
+import { StarkDOMUtil } from "@nationalbankbelgium/stark-ui";
+import { StarkMinimapItemProperties } from "./item-properties.intf";
+
+/**
+ * @ignore
+ */
+const _isEqual: Function = require("lodash/isEqual");
+
+/**
+ * @ignore
+ */
+const _cloneDeep: Function = require("lodash/cloneDeep");
+
+export type StarkMinimapComponentMode = "compact";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-minimap";
+
+/**
+ * Component to display a minimap which permits to toggle the visibility of passed elements.
+ * The minimap shows the label of the elements to display with a checkbox to enable/disable the visibility:
+ *
+ * @param string -mode Desired layout or flavour:
+ *
+ *    - compact: displayed in a compact form. There is only a button to trigger the dropdown.
+ *    - default: basic implementation with everything
+ *
+ * @param StarkMinimapItemProperties[] - items Array of StarkMinimapItemProperties objects which define the items to display in the minimap.
+ *
+ * @param string[] - visibleItems Array of names of the items that are visible.
+ *
+ * @param showHideItem EventEmitter to be called when call showHideItem callback.
+ *
+ */
+@Component({
+	selector: "stark-minimap",
+	templateUrl: "./minimap.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkMinimapComponent implements OnInit, OnChanges, OnDestroy {
+	@Input()
+	public mode?: StarkMinimapComponentMode;
+	@Input()
+	public parentItems: StarkMinimapItemProperties[];
+	@Input()
+	public parentVisibleItems: string[];
```

Comment:
Previously in Angular 1, we added the "parent' prefix as a workaround to avoid some issues with duplicated names, but that is not needed anymore in Angular2+... so you can rename this property to `visibleItems` 😉